src/factory-in-proc.js

```diff
@@ -65,10 +65,13 @@ class FactoryInProc {
       callback = options
       options = {}
     }
-    const IPFS = this.exec
-    new IPFS(options).version((err, _version) => {
-      if (err) { callback(err) }
-      callback(null, _version)
+
+    const node = new Node(options)
+    node.once('ready', () => {
+      node.init((err) => {
+        if (err) { return callback(err) }
+        node.version(callback)
+      })
```

Comment:
It is very odd to me that we start a new instance of IPFS to get the version. This will create a memory leak + it will lock the repo for a consequent node spawn. @dryajov can you expand why to do it this way?

Reply:
@diasdavid I'm not aware of any other way to get the version - is there a better one you can suggest?

Reply:
btw, the instance is not started, it is simply initialized with a repo, to start it we would have to call `node.start`.

Reply:
@dryajov are you saying that a repo is created to just get the version of ipfs? Sound like the node.version should work offline/un'init for the info it can provide, namely, the impl version.

If it was absolutely impossible (which it isn't), at least there should be a cleanup step here.

Reply:
Also, I thought about just using `ipfs-repo` directly to read the version, but it turns out we still need to initialize a repo to get the version, since there is no guarantee that the default repo in `~/.jsipfs` corresponds to the version of the IPFS in our `exec` param.

Reply:
Well, it turns out that the version is read by the repo here - https://github.com/ipfs/js-ipfs/blob/494da7f8c5b35f491f22a986ff5e8c456cc0e602/src/core/components/version.js#L14...L24. I think that's an overkill and we should separate the repo version from the ipfs version, that way we don't need a repo. If noone objects I can go ahead and make that change in `IPFS`.

Reply:
There is a need for a Repo version and IPFS version, that is correct. It should be ok to get the IPFS version without the Repo existing though.